src/compas_fab/robots/robot.py

```diff
@@ -911,6 +864,27 @@ def position_constraint_from_frame(self, frame_WCF, tolerance_position, group=No
         sphere = Sphere(frame_WCF.point, tolerance_position)
         return PositionConstraint.from_sphere(ee_link, sphere)
 
+    def position_constraint_from_max_reach(self, target_point, link_name, max_reach, group=None):
```

Comment:
I don't get the added value of this function. The only difference to `position_constraint_from_frame` is that you can pass the link name. If that is desired, than I'd suggest to rather add an optional parameter to `position_constraint_from_frame`.
Additionally, the scaling happens only before the planning request and should stay there.

If you found out that the way the scaling on a `Sphere` is wrong, please fix or raise an issue on COMPAS.

Reply:
I was also confused about this at first, but I was chatting with @matteo-pacher and I think it actually makes a lot of sense. The naming of these functions is still to be discussed, in few words, the goal of this was to build a constraint that uses a sphere to achieve an arm position that is fully stretched (ie close to singularity). I think it's a very clever approach to  tackle the problem.

@matteo-pacher perhaps you could expand this explanation on the PR description (maybe adding a few images) to show how this is supposed to work.